status/versions.go

```diff
@@ -187,15 +181,15 @@ type p8sResponseVersion struct {
 func jaegerVersion() (*ExternalServiceInfo, error) {
 	product := ExternalServiceInfo{}
 	product.Name = "Jaeger"
-	product.Url = config.Get().ExternalServices.Jaeger.URL
+	product.Url, _ = DiscoverJaeger()
```

Comment:
I wonder if DiscoverJaeger() should be a one time operation or invoked everytime that jaegerVersion() is called.
If I understand we call /status from the UI very often.
Perhaps this is a thing to be cached some way, as I don't think the location of jaeger/grafana will change to invoke the Discover on every request of the UI.

Reply:
mmm Maybe we should create another endpoint to force reload the status like /api/status/refresh and let this to be a one time operation

Reply:
+1 on caching that and not evaluating every time /status is called.
It could be from the UI called the moment the About... box is shown.

Reply:
Creating and maintaining a new endpoint sounds complex here, I mean, once Kiali has discovered there endpoints, probably they won't change unless some troubleshooting/demo where then probably a workaround could be to recreate kiali pod to refresh it.

I've just proposed something simple here, I mean, if we find some trouble with this design (we need to refresh that info) the suggested /status/refresh or similar operation can be implemented, but I don't see a reason to do it in a normal operation (and to recreate the pod sounds reasonable for a troubleshooting workaround).

Reply:
If the URLs are not set in config, we call the k8s API for this every time we call /status (if I get the code above). I don't think those additional calls every few seconds are really needed. 
But my Golang code reading-fu is not strong ..

Reply:
So at the moment it's set in config so not re-discovering every time.
Now the drawback is that the only way to refresh it (if the user changed grafana/jaeger url), kiali needs to be restarted. But I guess it's a reasonable-enough limitation.